### PR TITLE
feat: 891 - new "uploaded timestamp" field for raw images

### PR DESCRIPTION
**Pull request by monsieurtanuki**
### What
- Now we use the "uploaded timestamp" field that the server provided for raw images, but that we just didn't retrieve.
- With this field we'll be able to say if an uploaded image is "too old", e.g. if it was uploaded more than a year ago.

### Fixes bug(s)
- #891

### Part of 
- https://github.com/openfoodfacts/openfoodfacts-server/issues/9226
- https://github.com/openfoodfacts/smooth-app/issues/4674

### Impacted files:
* `api_json_to_from_test.dart`: added tests around new field `uploaded`
* `json_helper.dart`: added the `uploaded` field to from/to json conversion methods; refactored
* `product_image.dart`: added field `uploaded`, for uploaded images only

### CODE REVIEW

File: lib/src/model/product_image.dart
Comment (AshAman999) [inline lib/src/model/product_image.dart:133]: I think we can have a better variable name, maybe something like uploaded_on or created_at. Or is it the same var name in other places?

Comment (monsieurtanuki) [reply lib/src/model/product_image.dart:133]: Hi @AshAman999!

In this PR we have an `'uploaded_t'` JSON tag and an `int? uploaded` field.

If we compare to similar fields:
* we also have a `'created_t'` JSON tag and a `created` field (cf. in `Product`)
* but instead of `int?` in this PR, we do have a much more explicit `DateTime?`

Would you be OK with `DateTime? uploaded`? That would be consistent with the rest of the timestamp fields in the package, and wouldn't be confusing with a `DateTime?` type.

**Review by teolemon — APPROVED** [review]
per @AshAman999 's comment

**Comment by monsieurtanuki** [comment]
Thank you @teolemon @AshAman999 for your comments!
I don't merge right now as `DateTime? uploaded` was indeed a good idea instead of `int?`. I'll implement that by tomorrow.

**Comment by monsieurtanuki** [comment]
Thank you @teolemon @AshAman999: I've just merged the code with `DateTime? uploaded` field.

**Pull request merged**
